api/src/routes/auth.ts

```diff
@@ -93,28 +64,40 @@ router.use(function (req, res, next) {
     next();
 });
 
-router.get("/login", async function (req, res) {
+function saveReferer(req, res, next) {
     if (req.query.referer ?? false) {
         req.session.referer = req.query.referer;
     }
-    const user = await Database.getInstance().getUserBy("username", "chris");
-    res.render("../../views/login-test", { user });
-});
+    next();
+}
 
-router.get("/logout", function (req, res) {
-    req.logout();
-    res.redirect(Config.getInstance().clientUrl);
-});
+function showLoginForm(req, res) {
```

Comment:
Because it is used to establish a session on the API server, which is independent of any client that might be consuming the API, I think it makes sense to keep it on the API side.

Reply:
(Essentially this is the service that the React client delegates to in order to perform a login, when it hasn't yet acquired a token).